src/cljs/main/broadfcui/page/library/library_page.cljs

```diff
@@ -74,65 +81,63 @@
-                          " and request access for the "
-                          (:namespace data) "/" (:name data) " workspace."])})))}))
+     (if (:workspaceAccess data)
+       (nav/navigate (:nav-context props) "workspaces" (common/row->workspace-id data))
+       (comps/push-message
+        {:header "Request Access"
+         :message
+         (if (= (config/tcga-namespace) (:namespace data))
+           [:span {}
+            [:p {} "For access to TCGA protected data please apply for access via dbGaP [instructions can be found "
+             [:a {:href "https://wiki.nci.nih.gov/display/TCGA/Application+Process" :target "_blank"} "here"] "]."]
+            [:p {} "After dbGaP approves your application please link your eRA Commons ID in your FireCloud profile page."]]
+           [:span {}
+            "Please contact "
+            [:a {:target "_blank"
+                 :href (str "mailto:" (:library:contactEmail data))}
+             (str (:library:datasetCustodian data) " <" (:library:contactEmail data) ">")]
+            " and request access for the "
+            (:namespace data) "/" (:name data) " workspace."])})))
    :build-aggregate-fields
    (fn [{:keys [props]}]
      (reduce
-       (fn [results field] (assoc results field (if (contains? (:expanded-aggregates props) field) 0 5)))
-       {}
-       (:aggregate-fields props)))
+      (fn [results field] (assoc results field (if (contains? (:expanded-aggregates props) field) 0 5)))
+      {}
+      (:aggregate-fields props)))
    :pagination
    (fn [{:keys [this state props]}]
      (fn [{:keys [current-page rows-per-page]} callback]
        (when-not (empty? (:aggregate-fields props))
          (endpoints/call-ajax-orch
-           (let [from (* (- current-page 1) rows-per-page)]
-             {:endpoint endpoints/search-datasets
-              :payload {:searchString (:search-text props)
-                        :filters (utils/map-keys name (:facet-filters props))
-                        :from from
-                        :size rows-per-page
-                        :fieldAggregations (if (= 1 current-page)
-                                             (react/call :build-aggregate-fields this)
-                                             {})}
-              :headers utils/content-type=json
-              :on-done
-              (fn [{:keys [success? get-parsed-response status-text]}]
-                (if success?
-                  (let [{:keys [total results aggregations]} (get-parsed-response)]
-                    (swap! state assoc :total total)
-                    (callback {:group-count total
-                               :filtered-count total
-                               :rows results})
-                    (when (= 1 current-page)
-                      ((:callback-function props) aggregations)))
-                  (callback {:error status-text})))})))))})
+          (let [from (* (- current-page 1) rows-per-page)]
```

Comment:
ugh I always change the spacing of the entire page without meaning to... but this should be right? it's what intellij gave me..